File: src/main/java/Piruru/cards/FreezeEnemy.java
package Piruru.cards;

import Piruru.actions.FreezeMonsterAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

import static Piruru.Piruru.makeID;

public class FreezeEnemy extends PiruruCard {
    private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(
            makeID(FreezeEnemy.class.getSimpleName()));
    private static final int COST = 2;
    private static final CardType TYPE = CardType.SKILL;
    private static final CardRarity RARITY = CardRarity.UNCOMMON;
    private static final CardTarget TARGET = CardTarget.ENEMY;
    private static final int COST_UP = 1;

    public FreezeEnemy() {
        super(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, 0, COST_UP);
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        act(new FreezeMonsterAction(m, p));
    }
}

File: src/main/java/Piruru/cards/FreezeEnemy.kt
package Piruru.cards

import Piruru.Piruluk.Statics.makeID
import Piruru.abstracts.PiruruCard
import Piruru.actions.FreezeMonsterAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.localization.CardStrings
import com.megacrit.cardcrawl.monsters.AbstractMonster


class FreezeEnemy : PiruruCard(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, 0, COST_UP) {

    override fun use(p: AbstractPlayer, m: AbstractMonster) {
        act(FreezeMonsterAction(m, p))
    }

    companion object {
        private val cardStrings = CardCrawlGame.languagePack.getCardStrings(
                makeID(FreezeEnemy::class.java))
        private val COST = 2
        private val TYPE = AbstractCard.CardType.SKILL
        private val RARITY = AbstractCard.CardRarity.UNCOMMON
        private val TARGET = AbstractCard.CardTarget.ENEMY
        private val COST_UP = 1
    }
}